lib/memory_profiler.js

```diff
@@ -0,0 +1,88 @@
+const memwatch = require('memwatch-ng')
+const heapdump = require('heapdump')
+const knox = require('knox')
+const path = require('path')
+const os = require('os')
+const chalk = require('chalk')
+
+const { NODE_ENV } = process.env
+const { S3_KEY, S3_SECRET, S3_BUCKET } = require('../desktop/config')
+
+function log (message) {
+  console.log(chalk.red(`[MEMORY] ${message}`))
+}
+
+function upload (pathname, callback) {
+  const filename = path.basename(pathname)
+  const client = knox.createClient({ key: S3_KEY, secret: S3_SECRET, bucket: S3_BUCKET })
+  client.putFile(filename, pathname, (err, res) => {
```

Comment:
Yeah let's upload them to a heapdumps directory—it'll get rather noisy otherwise.